modules/core/src/main/scala/org/scalasteward/core/bitbucketserver/http4s/Http4sBitbucketServerApiAlg.scala

```diff
@@ -0,0 +1,157 @@
+)(implicit client: HttpJsonClient[F])
+    extends VCSApiAlg[F] {
+  val url = new StashUrls(bitbucketApiHost)
+
+  override def createFork(repo: Repo): F[RepoOut] = ni(s"createFork($repo)")
+
+  override def createPullRequest(repo: Repo, data: NewPullRequestData): F[PullRequestOut] = {
+    val fromRef =
+      Json.Ref("refs/heads/" + data.head, Json.Repository(repo.repo, Json.Project(repo.owner)))
+    val toRef =
+      Json.Ref("refs/heads/" + data.base.name, Json.Repository(repo.repo, Json.Project(repo.owner)))
+
+    val req =
+      Json.NewPR(
+        title = data.title,
+        description = data.body,
+        state = Open,
+        open = true,
+        closed = false,
+        fromRef = fromRef,
+        toRef = toRef,
+        locked = false,
+        reviewers = List.empty
+      )
+
+    client
+      .postWithBody[Json.PR, Json.NewPR](url.pullRequests(repo), req, modify(repo))
+      .map(pr => PullRequestOut(pr.links("self").head.href, pr.state, pr.title))
+  }
+
+  override def getBranch(repo: Repo, branch: Branch): F[BranchOut] =
+    ni(s"createBranch($repo,$branch)")
+
+  override def getRepo(repo: Repo): F[RepoOut] =
+    for {
+      r <- client.get[Json.Repo](url.repo(repo), modify(repo))
+      cloneUri = r.links("clone").find(_.name.contains("http")).get.href
+    } yield RepoOut(r.name, UserOut(user.login), None, cloneUri, Branch("master"))
+
+  override def listPullRequests(repo: Repo, head: String, base: Branch): F[List[PullRequestOut]] =
+    client
+      .get[Json.Page[Json.PR]](url.listPullRequests(repo, s"refs/heads/$head"), modify(repo))
+      .map(_.values.map(pr => PullRequestOut(pr.links("self").head.href, pr.state, pr.title)))
+
+  def ni(name: String): Nothing = throw new NotImplementedError(name)
+
+  object Json {
+
+    case class Page[A](values: List[A])
+
+    case class Repo(name: String, forkable: Boolean, project: Project, links: Links)
+
+    case class Project(key: String)
+
+    type Links = Map[String, NonEmptyList[Link]]
+
+    case class Link(href: Uri, name: Option[String])
+
+    case class PR(title: String, state: PullRequestState, links: Links)
+
+    case class NewPR(
+        title: String,
+        description: String,
+        state: PullRequestState,
+        open: Boolean,
+        closed: Boolean,
+        fromRef: Ref,
+        toRef: Ref,
+        locked: Boolean,
+        reviewers: List[Reviewer]
+    )
+
+    case class Ref(id: String, repository: Repository)
+
+    case class Repository(slug: String, project: Project)
+
+    case class Reviewer(user: User)
+
+    case class User(name: String)
+
+    implicit def pageDecode[A: Decoder]: Decoder[Page[A]] = deriveDecoder
+    implicit val repoDecode: Decoder[Repo] = deriveDecoder
+    implicit val projectDecode: Decoder[Project] = deriveDecoder
+    implicit val linkDecoder: Decoder[Link] = deriveDecoder
+    implicit val uriDecoder: Decoder[Uri] = Decoder.decodeString.map(Uri.unsafeFromString)
```

Comment:
@Milyardo This `Decoder` is unsafe, you should import [`util.uri.uriDecoder`](https://github.com/fthomas/scala-steward/blob/cae7c5793cf0bcffc0796ce70b2371cbe6388e8a/modules/core/src/main/scala/org/scalasteward/core/util/uri.scala#L26-L27) instead.

Reply:
Sure thing, apologies for the delay, I don't use github that often.